classes/image.php

```diff
@@ -87,10 +88,10 @@ public static function resize (
             return false;
         }
 
-        $contextid = $originalfile->get_contextid();
-        $component = $originalfile->get_component();
-        $filearea = $originalfile->get_filearea();
-        $itemid = $originalfile->get_itemid();
+        $contextid = $public ? 1 : $originalfile->get_contextid();
```

Comment:
just wondering if this would be clearer as an if else instead of 4 ternaries

Reply:
Personally, I think it reads fine as is.